vulnerabilities/pipelines/v2_importers/collect_repo_fix_commits.py

```diff
@@ -0,0 +1,117 @@
+    r"\bXSA-\d{1,4}\b",
+]
+
+
+class CollectRepoFixCommitPipeline(VulnerableCodeBaseImporterPipelineV2):
+    """
+    Pipeline to collect fix commits from any git repository.
+    """
+
+    pipeline_id = "repo_fix_commit"
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.clone,
+            cls.collect_and_store_advisories,
+            cls.clean_downloads,
+        )
+
+    def clone(self):
+        """Clone the repository."""
+        self.repo_url = "https://github.com/torvalds/linux"
+        repo_path = tempfile.mkdtemp()
+        cmd = [
+            "git",
+            "clone",
+            "--bare",
+            "--filter=blob:none",
+            "--no-checkout",
+            self.repo_url,
+            repo_path,
+        ]
+        subprocess.run(cmd, check=True)
+        self.repo = Repo(repo_path)
+
+    def advisories_count(self) -> int:
+        return int(self.repo.git.rev_list("--count", "HEAD"))
+
+    def classify_commit_type(self, commit) -> list[str]:
+        """
+        Extract vulnerability identifiers from a commit message.
+        Returns a list of matched vulnerability IDs (normalized to uppercase).
+        """
+        matches = []
+        for pattern in SECURITY_PATTERNS:
+            found = re.findall(pattern, commit.message, flags=re.IGNORECASE)
+            matches.extend(found)
+        return matches
+
+    def collect_fix_commits(self):
+        """
+        Iterate through repository commits and group them by vulnerability identifiers.
+        return a list with (vuln_id, [(commit_id, commit_message)]).
+        """
+        self.log("Processing git repository fix commits (grouped by vulnerability IDs).")
+
+        grouped_commits = defaultdict(list)
+        for commit in self.repo.iter_commits("--all"):
+            matched_ids = self.classify_commit_type(commit)
+            if not matched_ids:
+                continue
+
+            commit_id = commit.hexsha
+            commit_message = commit.message.strip()
+
+            for vuln_id in matched_ids:
+                grouped_commits[vuln_id].append((commit_id, commit_message))
+
+        self.log(f"Found {len(grouped_commits)} vulnerabilities with related commits.")
+        self.log("Finished processing all commits.")
+        return grouped_commits
+
+    def collect_advisories(self):
+        """
+        Generate AdvisoryData objects for each vulnerability ID grouped with its related commits.
+        """
+        self.log("Generating AdvisoryData objects from grouped commits.")
+        grouped_commits = self.collect_fix_commits()
+        for vuln_id, commits in grouped_commits.items():
+            references = [ReferenceV2(url=f"{self.repo_url}/commit/{cid}") for cid, _ in commits]
+
+            summary_lines = [f"- {cid}: {msg}" for cid, msg in commits]
+            summary = f"Commits fixing {vuln_id}:\n" + "\n".join(summary_lines)
+            yield AdvisoryData(
+                advisory_id=vuln_id,
```

Comment:
This will be problematic since we intend to collect fixed commits from multiple different repo here. Suppose we get a fix commit for `CVE-000-000` in two different repo, we will end up with a conflict while inserting the advisory, as we use `advisory_id` prefixed with the `pipeline_id` to create unique AVID. In this case we will end up with same AVID for fix commits imported from two different git repos.

Reply:
@keshav-space Not sure what the best solution for this is, but based on my understanding, we can make the pipeline_id dynamic

* `django_fix_commit`
* `django_restframework_fix_commit`

For example:

```yaml
avid: "e.g., django_fix_commit/PYSEC-2020-2233"
avid: "e.g., django_restframework_fix_commit/PYSEC-2020-2233"
```

This will generate a different `avid` for each Git repository.
I’m not sure if this completely solves the problem, though.